gluonnlp/data/transforms.py

```diff
@@ -177,20 +178,23 @@ def __init__(self):
                               'official installation guide in https://www.nltk.org/install.html .')
         self._tokenizer = MosesTokenizer()
 
-    def __call__(self, sample):
+    def __call__(self, sample, return_str=False):
         """
 
         Parameters
         ----------
         sample: str
             The sentence to tokenize
+        return_str: bool, default False
+            True: return a single string
```

Comment:
why would a tokenizer want to return a single string? does some cleaning happen in this process?

Reply:
This can be used if we want to write the tokenized text into a file.

Reply:
In nltk, return_str=true means the function ignores str.split().

Reply:
This flag assumes the separator token to use when generating this string. I think doing this in a separate transformer function is likely cleaner. Is it easy to implement the joining logic to mimic the behavior in moses tokenizer?

Reply:
this corresponds to a single line in nltk tokenzier: `return text if return_str else text.split()`

Reply:
so they assume that separator is white space, which is hard decoded.

Reply:
The simple split() already removes more than one white spaces, why it should be different? The nltk implements the same logic as in Mose tokenizer, so I think their function is reasonable.

Reply:
```
In [1]: from nltk.tokenize.moses import MosesTokenizer

In [2]: t = MosesTokenizer()

In [3]: x = ['a', ' ', ' ', 'b']

In [4]: xs = ''.join(x)

In [5]: xs
Out[5]: 'a  b'

In [6]: t.tokenize(xs)
Out[6]: ['a', 'b']

In [7]: t.tokenize(xs, return_str=True)
Out[7]: 'a b'
```

Reply:
note that the output is different from original string

Reply:
this is because Mose tokenizer will remove redundant space. So why it would be a problem? That's why we sometimes cannot recover the original string by calling tokenizer.perl and detokenizer.perl.